src/fft/fftn.rs

```diff
@@ -0,0 +1,534 @@
+///
+/// # Panic
+///
+/// - if the input array is a scalar array
+/// - if the axes are not unique
+/// - if the shape and axes have different sizes
+/// - if the output sizes are invalid (<= 0)
+/// - if more axes are provided than the array has
+///
+/// See [`try_fftn_device`] for more details.
+#[default_device(device = "cpu")]
+pub fn fftn_device<'a>(
+    a: &'a Array,
+    s: impl Into<Option<&'a [i32]>>,
+    axes: impl Into<Option<&'a [i32]>>,
+    stream: StreamOrDevice,
+) -> Array {
+    try_fftn_device(a, s, axes, stream).unwrap()
+}
+
+/// One dimensional inverse discrete Fourier Transform.
+///
+/// # Params
+///
+/// - `a`: Input array.
+/// - `n`: Size of the transformed axis. The corresponding axis in the input is truncated or padded
+///   with zeros to match `n`. The default value is `a.shape[axis]` if not specified.
+/// - `axis`: Axis along which to perform the FFT. The default is `-1` if not specified.
+#[default_device(device = "cpu")]
+pub unsafe fn ifft_device_unchecked(
+    a: &Array,
+    n: impl Into<Option<i32>>,
+    axis: impl Into<Option<i32>>,
+    stream: StreamOrDevice,
+) -> Array {
+    let (n, axis) = resolve_size_and_axis_unchecked(a, n, axis);
+    unsafe {
+        let c_array = mlx_sys::mlx_fft_ifft(a.c_array, n, axis, stream.stream.c_stream);
+        Array::from_ptr(c_array)
+    }
+}
+
+/// One dimensional inverse discrete Fourier Transform.
+///
+/// # Params
+///
+/// - `a`: Input array.
+/// - `n`: Size of the transformed axis. The corresponding axis in the input is truncated or padded
+///  with zeros to match `n`. The default value is `a.shape[axis]` if not specified.
+/// - `axis`: Axis along which to perform the FFT. The default is `-1` if not specified.
+#[default_device(device = "cpu")]
+pub fn try_ifft_device(
+    a: &Array,
+    n: impl Into<Option<i32>>,
+    axis: impl Into<Option<i32>>,
+    stream: StreamOrDevice,
+) -> Result<Array, FftError> {
+    let (n, axis) = try_resolve_size_and_axis(a, n, axis)?;
+    unsafe { Ok(ifft_device_unchecked(a, n, axis, stream)) }
+}
+
+/// One dimensional inverse discrete Fourier Transform.
+///
+/// # Params
+///
+/// - `a`: Input array.
+/// - `n`: Size of the transformed axis. The corresponding axis in the input is truncated or padded
+///  with zeros to match `n`. The default value is `a.shape[axis]` if not specified.
+/// - `axis`: Axis along which to perform the FFT. The default is `-1` if not specified.
+#[default_device(device = "cpu")]
+pub fn ifft_device(
+    a: &Array,
+    n: impl Into<Option<i32>>,
+    axis: impl Into<Option<i32>>,
+    stream: StreamOrDevice,
+) -> Array {
+    try_ifft_device(a, n, axis, stream).unwrap()
+}
+
+/// Two dimensional inverse discrete Fourier Transform.
+///
+/// # Params
+///
+/// - `a`: The input array.
+/// - `s`: Size of the transformed axes. The corresponding axes in the input are truncated or padded
+/// with zeros to match `n`. The default value is the sizes of `a` along `axes`.
```

Comment:
```suggestion
/// with zeros to match `s`. The default value is the sizes of `a` along `axes`.
```

Reply:
This follows the python api. 1D fft uses n, n-D uses s